File: AndroidCaseStudy/Fundamental/LifecycleActivityFragment/androidactivityfragment_one/src/main/java/com/algokelvin/lifecycle/TwoFragment.java
package com.algokelvin.lifecycle;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import org.jetbrains.annotations.NotNull;

public class TwoFragment extends Fragment {
    private OnDataPass onDataPass;
    private String data2;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_two, container, false);
    }

    @Override
    public void onAttach(@NonNull @org.jetbrains.annotations.NotNull Context context) {
        super.onAttach(context);
        onDataPass = (OnDataPass) context;
    }

    @Override
    public void onViewCreated(@NonNull @NotNull View view, @Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        TextView txtFragment = view.findViewById(R.id.txt_fragment_two);
        this.data2 = txtFragment.getText().toString();
        passData();

    }

    public void passData() {
        onDataPass.onDataPass(data2);
    }

}

File: AnotherAndroidApp/app-fragment-01/src/main/java/algorithm/kelvin/app/fragment_01/TwoFragment.kt
package algorithm.kelvin.app.fragment_01

import android.content.Intent
import android.os.Bundle
import android.util.Log
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.fragment_two.*

class TwoFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_two, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        btnTwoFromFragmentTwo.setOnClickListener {
            startActivity(Intent(requireContext(), TwoActivity::class.java))
        }
    }

    override fun onResume() {
        super.onResume()
        Log.i("fragment-resume", "Resume 2 is start")
    }

    override fun onStart() {
        super.onStart()
        Log.i("fragment-start", "Start 2 is start")
    }

    override fun onStop() {
        super.onStop()
        Log.i("fragment-stop", "Stop 2 is start")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.i("fragment-destroy", "Destroy 2 is start")
    }

}
